astropy/wcs/utils.py

```diff
@@ -1077,9 +1077,16 @@ def fit_wcs_from_points(xy, world_coords, proj_point='center',
     # fit linear terms, assign to wcs
     # use (1, 0, 0, 1) as initial guess, in case input wcs was passed in
     # and cd terms are way off.
+    # Use bounds to require that the fit center pixel is on the input image
+    xpmin, xpmax, ypmin, ypmax = xp.min(), xp.max(), yp.min(), yp.max()
+    if xpmin==xpmax: xpmin, xpmax = xpmin-0.5, xpmax+0.5
+    if ypmin==ypmax: ypmin, ypmax = ypmin-0.5, ypmax+0.5
```

Comment:
Please split the above two lines so it's consistent with the rest of the package.